src/main/java/org/broadinstitute/hellbender/tools/walkers/sv/SVAnnotateEngine.java

```diff
@@ -0,0 +1,734 @@
+                                                                   final SVIntervalTree<String> nonCodingIntervalTree,
+                                                                   final Set<String> MSVExonOverlapClassifications,
+                                                                   final SAMSequenceDictionary sequenceDictionary,
+                                                                   final int maxBreakendLen) {
+        final Map<String, Set<String>> variantConsequenceDict = new HashMap<>();
+        final StructuralVariantAnnotationType overallSVType = getSVType(variant);
+        final List<SVSegment> svSegments = getSVSegments(variant, overallSVType, maxBreakendLen);
+        if (gtfIntervalTree != null) {
+            for (SVSegment svSegment : svSegments) {
+                annotateGeneOverlaps(svSegment.getInterval(), svSegment.getIntervalSVType(), variantConsequenceDict,
+                        MSVExonOverlapClassifications, sequenceDictionary, gtfIntervalTree);
+            }
+        }
+
+        // if variant consequence dictionary is empty (no protein-coding annotations), apply INTERGENIC flag
+        final boolean noCodingAnnotations = variantConsequenceDict.isEmpty();
+
+        // then annotate promoter overlaps and non-coding feature overlaps
+        if (promoterIntervalTree != null) {
+            for (final SVSegment svSegment : svSegments) {
+                annotatePromoterOverlaps(svSegment.getInterval(), variantConsequenceDict, promoterIntervalTree,
+                        sequenceDictionary);
+            }
+        }
+
+        if (nonCodingIntervalTree != null) {
+            for (SVSegment svSegment : svSegments) {
+                annotateNonCodingOverlaps(svSegment.getInterval(), variantConsequenceDict, nonCodingIntervalTree,
+                        sequenceDictionary);
+            }
+        }
+
+        // annotate nearest TSS for intergenic variants with no promoter overlaps
+        if (transcriptionStartSiteTree != null && !variantConsequenceDict.containsKey(GATKSVVCFConstants.PROMOTER) &&
+                noCodingAnnotations) {
+            for (SVSegment svSegment : svSegments) {
+                annotateNearestTranscriptionStartSite(svSegment.getInterval(), variantConsequenceDict,
+                        transcriptionStartSiteTree, sequenceDictionary);
+            }
+        }
+
+        final Map<String, Object> attributes = sortVariantConsequenceDict(variantConsequenceDict);
+        if (gtfIntervalTree != null) {
+            attributes.put(GATKSVVCFConstants.INTERGENIC, noCodingAnnotations);
+        }
+        return attributes;
+    }
+
+    /**
+     * Create VariantContext for input variant with added functional annotation INFO keys
+     * @param variant - input VCF record
+     * @return - VariantContext equal to input + functional annotation INFO keys
+     */
+    protected VariantContext createAnnotatedStructuralVariantContext(final VariantContext variant) {
```

Comment:
```suggestion
    public VariantContext createAnnotatedStructuralVariantContext(final VariantContext variant) {
```
Otherwise there's no way to use this class outside its package